pkg/connection/connection.go

```diff
@@ -1,251 +0,0 @@
-/*
-Copyright 2018 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package connection
-
-import (
-	"context"
-	"fmt"
-
-	"github.com/container-storage-interface/spec/lib/go/csi"
-	"github.com/golang/glog"
-	"github.com/golang/protobuf/ptypes"
-	"github.com/golang/protobuf/ptypes/timestamp"
-	"github.com/kubernetes-csi/csi-lib-utils/connection"
-	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
-	"google.golang.org/grpc"
-	"k8s.io/api/core/v1"
-)
-
-// CSIConnection is gRPC connection to a remote CSI driver and abstracts all
-// CSI calls.
-type CSIConnection interface {
-	// GetDriverName returns driver name as discovered by GetPluginInfo()
-	// gRPC call.
-	GetDriverName(ctx context.Context) (string, error)
-
-	// SupportsControllerCreateSnapshot returns true if the CSI driver reports
-	// CREATE_DELETE_SNAPSHOT in ControllerGetCapabilities() gRPC call.
-	SupportsControllerCreateSnapshot(ctx context.Context) (bool, error)
-
-	// SupportsControllerListSnapshots returns true if the CSI driver reports
-	// LIST_SNAPSHOTS in ControllerGetCapabilities() gRPC call.
-	SupportsControllerListSnapshots(ctx context.Context) (bool, error)
```

Comment:
We still need this.

Reply:
I didn't see this get called anywhere though, should we still keep it?

Reply:
I can re-implement and add this check as part of `main.go` in a follow-up PR if that's okay

Reply:
Sure.